cmd/main.go

```diff
@@ -298,3 +330,35 @@ func loadCertificate(cfg config.MQTTConf) (config.MQTTConf, error) {
 	cfg.CA = caByte
 	return c, nil
 }
+
+func mergeConfigs(dst, src interface{}) interface{} {
+	d := reflect.ValueOf(dst).Elem()
+	s := reflect.ValueOf(src).Elem()
+
+	for i := 0; i < d.NumField(); i++ {
+		dField := d.Field(i)
+		sField := s.Field(i)
+		switch dField.Kind() {
+		case reflect.Struct:
```

Comment:
Why are we doing this reflection and not just simply parsing a TOML file? This looks too complicated, maybe some design is wrong and this can be simplified by simpler config file or some other external decision?